File: src/main/java/com/googongill/aditory/domain/ProfileImage.java
package com.googongill.aditory.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ProfileImage extends BaseTimeEntity {
    @Id @Column(name = "profile_image_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String originalName;
    private String uploadedName;

    public ProfileImage(String originalName, String uploadedName) {
        this.originalName = originalName;
        this.uploadedName = uploadedName;
    }
}

File: src/main/java/com/googongill/aditory/domain/ProfileImage.kt
package com.googongill.aditory.domain

import jakarta.persistence.*

@Entity
class ProfileImage(
        var originalName: String,
        var uploadedName: String
) : BaseTimeEntity() {
    @Id
    @Column(name = "profile_image_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    fun updateProfileImage(profileImage: ProfileImage) {
        originalName = profileImage.originalName
        uploadedName = profileImage.uploadedName
    }
}